### PR TITLE
Support fallback to plain http in nerdctl login

### LINKED ISSUE #887: http registry login (`http: server gave HTTP response to HTTPS client`)
I found this question #224 but setting the `--insecure-registry` flag only skips tls verfiy and does not affect the https://github.com/containerd/nerdctl/blob/354d03acb11a1649f0f3aec73218b8d8e2d9e062/pkg/imgutil/dockerconfigresolver/dockerconfigresolver.go#L39 option (anymore?).

I figured out plainHTTP is only accepted on localhost. That's totally fine in terms of security not sending credentials in plain text over the network. Nevertheless I have to do it (even though VPN) ...

Can we provide a new flag for http? Or could we parse the protocol form the host input for example `nerdctl login http://registry.com` ?

### PR DESCRIPTION
**Pull request by Zheaoli**
Fix #887 

Signed-off-by: Zheao.Li <me@manjusaka.me>

### CODE REVIEW

File: cmd/nerdctl/login.go [hunk 1]
```diff
@@ -240,14 +241,31 @@ func loginClientSide(ctx context.Context, cmd *cobra.Command, auth types.AuthCon
...
 	for i, rh := range regHosts {
 		err = tryLoginWithRegHost(ctx, rh)
 		identityToken := fetchedRefreshTokens[rh.Host] // can be empty
 		if err == nil {
 			return identityToken, nil
 		}
+		if insecure && (errutil.IsErrHTTPResponseToHTTPSClient(err) || errutil.IsErrConnectionRefused(err)) {
+			needFallbackToPlainHttp = true
+			needFallBackRegHosts = append(needFallBackRegHosts, rh)
+		}
```
Comment (Junnplus) [inline cmd/nerdctl/login.go:255]: I reserve my opinion on https://github.com/containerd/nerdctl/pull/894#discussion_r827661658.

When it is not a valid HTTPs request and insecure-registry flaged, we can fallback directly to plain HTTP. cc @fahedouch

```suggestion
                err = tryLoginWithRegHost(ctx, rh)
		if err != nil && insecure && (errutil.IsErrHTTPResponseToHTTPSClient(err) || errutil.IsErrConnectionRefused(err)) {
		    rh.Scheme = "http"
		    err = tryLoginWithReghost(ctx, rh)
		}
		identityToken := fetchedRefreshTokens[rh.Host] // can be empty
		if err == nil {
			return identityToken, nil
		}
```

Comment (Zheaoli) [reply cmd/nerdctl/login.go:255]: OK, make sense

**Review by Junnplus — APPROVED** [review]
LGTM. Thanks!

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**